lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
```

Comment:
@jcheng5, oh, I think my issues with client reconnects are due to a Connect limitation. I committed some changes that introduce a new `disconnect` decorator, but I'm still unclear on whether it's correct. I built this version and tested it with shiny server, and both server and client reconnects appear to work fine with it.